### PR TITLE
[target.d] refactor 64bit logic

### PR DESCRIPTION
**Pull request by thewilsonator**
reduce diff on Aarch64 backend: https://github.com/dlang/dmd/pull/16554

### CODE REVIEW

File: compiler/src/dmd/target.d [hunk 1]
```diff
@@ -363,7 +363,7 @@ extern (C++) struct Target
     bool isX86_64;          // generate 64 bit code for x86_64; true by default for 64 bit dmd
     bool isX86;             // generate 32 bit Intel x86 code
     bool isLP64;            // pointers are 64 bits
-
+    extern(D) bool is64() const @safe { return isX86_64 /*|| isAArch64*/; }
```
Comment (ibuclaw) [inline compiler/src/dmd/target.d:366]: It looks like this is being turned into being synonymous with `isLP64` -> use that field instead and remove this function?

Comment (WalterBright) [reply compiler/src/dmd/target.d:366]: With the rather confusing collection of nomenclature here, I am concerned that `is64` means "64 bit" and that implies that it includes `AArch64`. I'd rather stick with `X86_64` and `AArch64` explicitly, as those correspond to exact architecture names.

Comment (ljmf00) [reply compiler/src/dmd/target.d:366]: Yeah, this broke DMD as a library backthen, and `isLP64` is now the way to get that. I would suggest name everything the same

Comment (thewilsonator) [reply compiler/src/dmd/target.d]: > It looks like this is being turned into being synonymous with isLP64 -> use that field instead and remove this function?

Done.

**Review by RazvanN7 — APPROVED** [review]

**Comment by WalterBright** [comment]
I object to this because nobody knows what isLP64 actually means. I know it's _supposed_ to mean 64 bit pointers. But, in usage, it gets conflated with "are registers 64 bits". When starting the AArch64 support, I noticed a fair amount of confused use of it in the source code.

So, I switched it to x86_64 and AArch64, and tried to use them completely consistently in the source code. Those both have very specific meanings - 64 bit registers and 64 bit pointers. They are industry standard terms. I refactored all use of them to be consistent with this.

isLP64 is like strcpy to me. Always I have to stop and check if it is used properly. I have to always check if it is set properly. It sucks up time again and again. When I see `isX86_64 || isAArch64` I don't have to check.

A refactoring that would help would be to change `target.isAArch64` to `target.AArch64` so that it exactly matches the predefined version for it (I've already made this change in the backend). I'd prefer to rename `X86_64` usage to `x86_64` as well, as that capitalization is the industry standard casing. (Unfortunately, X86_64 is the version condition.)

It's better to not have to mentally do these translations all the time.

I also have long experience with the tangle of overlapping and constantly misused architecture #defines in the C DOS/Windows/Linux world, and really do not want to relive it.

**Pull request merged**